modules/components/analytics/config/org.wso2.carbon.pc.analytics.config/src/main/java/org/wso2/carbon/pc/analytics/config/utils/DASConfigurationUtils.java

```diff
@@ -0,0 +1,115 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.pc.analytics.config.utils;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.context.RegistryType;
+import org.wso2.carbon.pc.analytics.config.AnalyticsConfigConstants;
+import org.wso2.carbon.registry.api.Registry;
+import org.wso2.carbon.registry.api.RegistryException;
+import org.wso2.carbon.registry.api.Resource;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLStreamException;
+import java.io.File;
+import java.io.IOException;
+
+
+/**
+ * Utils file for DAS configuration related properties
+ */
+public class DASConfigurationUtils {
+    private static final Log log = LogFactory.getLog(DASConfigurationUtils.class);
+
+    public static void setPropertyDASAnalyticsConfigured(String processName, String processVersion) {
+        PrivilegedCarbonContext context = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        Registry registry = context.getRegistry(RegistryType.SYSTEM_GOVERNANCE);
+        Resource resource;
+        String processAssetPath = "processes/" + processName + "/" + processVersion;
+
+        try {
+            if (registry.resourceExists(processAssetPath)) {
+                resource = registry.get(processAssetPath);
+                if (resource.getProperty("isDASConfiguredForAnalytics") == null) {
+                    resource.addProperty("isDASConfiguredForAnalytics", "true");
+                    registry.put(processAssetPath, resource);
+                }
+            }
+        } catch (RegistryException e) {
+            log.error("Error working with SYSTEM_GOVERNANCE registry property -isDASConfiguredForAnalytics");
+        }
+    }
+
+    public static boolean isDASAnalyticsConfigured(String processName, String processVersion) {
+        PrivilegedCarbonContext context = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        Registry registry = context.getRegistry(RegistryType.SYSTEM_GOVERNANCE);
+        Resource resource;
+        String processAssetPath = "processes/" + processName + "/" + processVersion;
+
+        try {
+            if (registry.resourceExists(processAssetPath)) {
+                resource = registry.get(processAssetPath);
+                return Boolean.parseBoolean(resource.getProperty("isDASConfiguredForAnalytics"));
+            } else {
+                return false;
+            }
+        } catch (RegistryException e) {
+            log.error("Error in getting SYSTEM_GOVERNANCE registry property- isDASConfiguredForAnalytics ");
+        }
```

Comment:
shouldn't the exception be thrown. Here if a registry exception occurs, we are logging the error and then returning true. IMO this is wrong.